src/Microsoft.AspNetCore.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -32,7 +32,7 @@ public class InputTagHelper : TagHelper
                 { "Url", "url" },
                 { "EmailAddress", "email" },
                 { "Date", "date" },
-                { "DateTime", "datetime" },
+                { "DateTime", "datetime-local" },
```

Comment:
There are 3 cases and I thought you wanted to change only (1) below. @jbagga's changes cover (1) and (2). If that's what you want, her source changes are fine and the main bit is confirming all 3 cases are covered for HTML and tag helpers.
1. Property has type `DateTime` or `DateTimeOffset`.
2. Property has some random type and an associated `[DataType("datetime")]`.
3. Property has whatever type but user either entered `<input type="datetime" ...>` or included `[type]: "datetime"` in their `htmlAttributes` dictionary.